sql/core/src/main/scala/org/apache/spark/sql/execution/SparkQl.scala

```diff
@@ -42,6 +45,83 @@ private[sql] class SparkQl(conf: ParserConf = SimpleParserConf()) extends Cataly
           getClauses(Seq("TOK_QUERY", "FORMATTED", "EXTENDED"), explainArgs)
         ExplainCommand(nodeToPlan(query), extended = extended.isDefined)
 
+      case Token("TOK_REFRESHTABLE", nameParts :: Nil) =>
+        val tableIdent = extractTableIdent(nameParts)
+        RefreshTable(tableIdent)
+
+      case Token("TOK_CREATETABLEUSING", createTableArgs) =>
+        val clauses = getClauses(
+            Seq("TEMPORARY", "TOK_IFNOTEXISTS", "TOK_TABNAME", "TOK_TABCOLLIST",
+              "TOK_TABLEPROVIDER", "TOK_TABLEOPTIONS", "TOK_QUERY"), createTableArgs)
+
+        val temp = clauses(0)
+        val allowExisting = clauses(1)
+        val Some(tabName) = clauses(2)
+        val tableCols = clauses(3)
+        val Some(tableProvider) = clauses(4)
+        val tableOpts = clauses(5)
+        val tableAs = clauses(6)
+
+        val tableIdent: TableIdentifier = tabName match {
+          case Token("TOK_TABNAME", Token(dbName, _) :: Token(tableName, _) :: Nil) =>
+            new TableIdentifier(tableName, Some(dbName))
+          case Token("TOK_TABNAME", Token(tableName, _) :: Nil) =>
+            TableIdentifier(tableName)
+        }
+
+        val columns = tableCols.map {
+          case Token("TOK_TABCOLLIST", fields) => StructType(fields.map(nodeToStructField))
+        }
+
+        val provider = tableProvider match {
+          case Token("TOK_TABLEPROVIDER", Token(provider, _) :: Nil) => provider
```

Comment:
The ```TOK_TABLEPROVIDER``` will have multiple children if the name is dotted, e.g.:```org.apache.spark.sql.avro```. So we need to handle that case here.